task/mongo-optimization-task.js

```diff
@@ -240,25 +201,43 @@ async function task_3_1(db) {
                 ]
             }
         },
+        {
+            "$project" : {
+                "_id" : 1,
+                "contacts.id" : 1,
+                "contacts.questions.criteria_value" : 1,
+                "criteria_value" : {
+                    "$ifNull" : [
+                        "$contacts.questions.criteria_value",
+                        "$contacts.questions.answers.criteria_value"
+                    ]
+                },
+                "contacts.questions.id" : 1,
+                "contacts.questions.answers" : 1,
+                "contacts.questions.category_id" : 1
+            }
+        },
         {
             "$lookup" : {
                 "from" : "clientCriteria",
-                "localField" : "criteria_value",
-                "foreignField" : "value",
+                "let": {"criteria_Value": "$criteria_value"},
+                "pipeline": [
+                    {
+                        "$match" : {"versions.initiativeId" : ObjectId("58af4da0b310d92314627290")}
+                    },
+                    {
+                        "$match": {$expr: {$eq: ["$value",  "$$criteria_Value"]}}
+                    },
+                    {
+                        "$unwind" : "$versions"
```

Comment:
That's breaking change. It works by coincidence, but we had match after unwind also. By the logic it takes `$criteria.versions.definition` from specific version by `initiativeId`.

So you need one more `match` here also you can play with `$limit: 1` at the end of middle of the pipeline - it can help too.

Reply:
i also take $criteria.versions.definition from specific version by initiativeId in start of pipeline. 
Joined only:
  "$match" : {"versions.initiativeId" : ObjectId("58af4da0b310d92314627290")}
I'm leaving only:
"(add "$project" in last commit)"
  "$match": {$expr: {$eq: ["$value",  "$$criteria_Value"]}}
and deconstructs an array
"$unwind"
maybe I didn't understand something, But what am I doing wrong

Reply:
![image](https://user-images.githubusercontent.com/5592397/92969240-409bc700-f485-11ea-9539-1fbdbe2e4b5e.png)